wireguard/wireguard_freebsd.go

```diff
@@ -1,98 +1,157 @@
+		if _, err := ncutils.RunCmd(ifconfig+" "+nc.Name+" destroy", true); err != nil {
+			return err
 		}
 	}
-	for _, node := range config.GetNodes() {
-		var address netlink.Addr
-		var address6 netlink.Addr
-		address.IPNet = &node.Address
-		if address.IPNet.IP != nil {
-			logger.Log(3, "adding address ", address.IP.String(), "to netmaker address")
-			if err := netlink.AddrAdd(l, &address); err != nil {
-				logger.Log(0, "error adding addr", err.Error())
-				return err
+	if _, err := ncutils.RunCmd(ifconfig+" wg create name "+nc.Name, true); err != nil {
+		return err
+	}
+	//config
+	if err := strip(); err != nil {
+		return err
+	}
+	if err != nil {
+		return err
+	}
+	if _, err := ncutils.RunCmd(fmt.Sprintf("%s setconf %s %s", wg, nc.Name, os.TempDir()+"/netmaker.conf"), true); err != nil {
+		return err
+	}
+	if err := os.Remove(os.TempDir() + "/netmaker.conf"); err != nil {
+		return err
+	}
+	//add addresses
+	for _, address := range nc.Addresses {
+		if address.IP.To4() != nil {
+			if _, err := ncutils.RunCmd(ifconfig+" "+nc.Name+" inet "+address.IP.String()+" alias", true); err != nil {
+				return fmt.Errorf("error adding address to interface %w", err)
 			}
-		}
-		address6.IPNet = &node.Address6
-		if address6.IPNet.IP != nil {
-			logger.Log(3, "adding address", address6.IP.String(), "to netmaker interface")
-			err = netlink.AddrAdd(l, &address6)
-			if err != nil {
-				logger.Log(0, "error adding addr", err.Error())
-				return err
+		} else {
+			if _, err := ncutils.RunCmd(ifconfig+" "+nc.Name+" inet6 "+address.IP.String()+" alias", true); err != nil {
+				return fmt.Errorf("error adding address to interface %w", err)
 			}
 		}
 	}
+	//set MTU
+	if _, err := ncutils.RunCmd(ifconfig+" "+nc.Name+" mtu "+strconv.Itoa(nc.MTU), true); err != nil {
+		return fmt.Errorf("error setting mtu %w", err)
+	}
+	if _, err := ncutils.RunCmd(ifconfig+" "+nc.Name+" up", true); err != nil {
+		return fmt.Errorf("error bringing up interface %w", err)
+	}
 	return nil
 }
 
-// == private ==
-
-type netLink struct {
-	attrs *netlink.LinkAttrs
+func strip() error {
+	in, err := os.Open(config.GetNetclientPath() + "netmaker.conf")
+	if err != nil {
+		return err
+	}
+	defer in.Close()
+	out, err := os.Create(os.TempDir() + "/netmaker.conf")
+	if err != nil {
+		return err
+	}
+	defer out.Close()
+	writer := bufio.NewWriter(out)
+	scan := bufio.NewScanner(in)
+	scan.Split(bufio.ScanLines)
+	for scan.Scan() {
+		line := scan.Text()
+		if strings.Contains(strings.ToLower(line), "address") {
+			continue
+		}
+		if strings.Contains(strings.ToLower(line), "mtu") {
+			continue
+		}
+		if strings.Contains(strings.ToLower(line), "dns") {
+			continue
+		}
+		fmt.Fprint(writer, line+"\n")
+	}
+	writer.Flush()
+	return nil
 }
 
-func (nc *NCIface) getKernelLink() *netLink {
-	linkAttrs := netlink.NewLinkAttrs()
-	linkAttrs.Name = nc.Name
-	return &netLink{
-		attrs: &linkAttrs,
-	}
+func apply(n *config.Node, c *wgtypes.Config) error {
+	return nil
```

Comment:
looks like this func is not doing any operations, can be removed right?

Reply:
it is being used to overwrite the operation of the main wireguard.go file